src/main/java/com/spotify/autoscaler/Autoscaler.java

```diff
@@ -105,19 +111,20 @@ private void runForCluster(
     try {
       if (hasRun.putIfAbsent(cluster.clusterName(), true) == null) {
         session = BigtableUtil.createSession(cluster.instanceId(), cluster.projectId());
-        autoscaleJob.run(cluster, session, Instant::now);
+        makeAutoscaleJob(stackDriverClient, database, autoscalerMetrics)
```

Comment:
why do we even have the `makeAutoscaleJob` method? can't we just have `new AutoscaleJob(stackDriverClient, database, autoscalerMetrics);` here instead?

Reply:
It's difficult to mock constructors, you would need to use `PowerMockito` or something like that.